[Zhang Xuan] Duke Increments

zhangxuan97 opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

j-lum committed:
Change file mode on `gradle` to be executable

j-lum committed:
Merge pull request #12 from j-lum/gradle+x

Change file mode on `gradle` to be executable (#9)

zhangxuan97 committed:
Duke: add Level-1

zhangxuan97 committed:
Duke: add Level-2

zhangxuan97 committed:
Duke: add Level-3

zhangxuan97 committed:
Duke: add Todo, Deadline, Event classes

zhangxuan97 committed:
Duke: add Level-4

zhangxuan97 committed:
add TextUiTesting

zhangxuan97 committed:
Add separate Java files for each class

zhangxuan97 committed:
Duke: added some Exceptions

zhangxuan97 committed:
Duke: add Level-5

zhangxuan97 committed:
Duke: add Level-6

zhangxuan97 committed:
Duke: fixed level 5 syntax error

zhangxuan97 committed:
Merge branch 'master1'

zhangxuan97 committed:
Minimal modifications to follow coding standard, added some Javadoc comments

zhangxuan97 committed:
Update deadline, event, task, todo

zhangxuan97 committed:
Add level-7 to branch-level-7

zhangxuan97 committed:
Add level-8 to branch-level-8

zhangxuan97 committed:
Merge branch 'branch-Level-7'

zhangxuan97 committed:
Merge branch 'branch-Level-8'

zhangxuan97 committed:
Add duke/data/duke.txt

zhangxuan97 committed:
Add more OOP

zhangxuan97 committed:
Refactor into packages

zhangxuan97 committed:
Add J-Unit Testing

zhangxuan97 committed:
Removed debug message

zhangxuan97 committed:
Javadoc for main and command packages

zhangxuan97 committed:
Javadoc for task package

zhangxuan97 committed:
Update to abide by coding standards

zhangxuan97 committed:
Add find by keyword function

zhangxuan97 committed:
Merge branch 'branch-A-JavaDoc'

zhangxuan97 committed:
Merge branch 'branch-A-CodingStandard' into master

zhangxuan97 committed:
Merge branch 'branch-Level-9' into master

zhangxuan97 committed:
Modified javadoc to include line between method description and params, removed hard coding for file path

TeoShyanJie commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,58 @@
+import command.Command;
+import main.*;
```
I think maybe can try to import the specific task

zhangxuan97 replied:
Thanks for pointing it out 👍

TeoShyanJie commented on src/main/java/command/AddCommand.java:
```diff
@@ -0,0 +1,81 @@
+import main.Storage;
+import main.TaskList;
+import main.Ui;
+import task.*;
```
I think maybe try to import specific task class

TeoShyanJie reviewed (changes requested):
I have left some comment regarding the import of class. I think the overall program is really good and have a good OOP.

zhangxuan97 committed:
Update Duke: close Scanner

zhangxuan97 committed:
Merge branch 'gradle'

zhangxuan97 committed:
Update build.gradle: admit input from system.in

zhangxuan97 committed:
Complete checkstyle

zhangxuan97 committed:
Merge branch 'branch-A-CheckStyle'

zhangxuan97 committed:
Complete tutorial 1 of java FX

zhangxuan97 committed:
Complete tutorial 2 of java FX

zhangxuan97 committed:
Java FX Tutorial 3: Iteration 1

zhangxuan97 committed:
Java FX Tutorial 3: Iteration 2

zhangxuan97 committed:
Java FX Tutorial 3: Iteration 3

Auxinnn commented on src/main/java/command/DoneCommand.java:
```diff
@@ -0,0 +1,41 @@
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        if (this.taskID > tasks.size()) {
+            throw new DukeException("Invalid ask ID. Please Enter a task ID between 1 and " + tasks.size());
```
Does 'ask' means 'task'?

zhangxuan97 replied:
Oh yea it does! Thanks for pointing it out 👍

Auxinnn reviewed (commented):
I think your codes are really good with nice abstraction.

zhangxuan97 committed:
Incorporate GUI

zhangxuan97 committed:
Merge branch 'branch-Level-10'

zhangxuan97 committed:
Small changes

zhangxuan97 committed:
Update to include fxml support

zhangxuan97 committed:
Merge branch 'branch-FXML'

zhangxuan97 committed:
Add shadowjar

zhangxuan97 committed:
remove filepath hardcoding

zhangxuan97 committed:
Add assertions

zhangxuan97 committed:
Merge pull request #5 from zhangxuan97/branch-A-Assertions

Add assertions

zhangxuan97 committed:
Merge branch 'master' of https://github.com/zhangxuan97/duke

zhangxuan97 committed:
coding quality updates

zhangxuan97 committed:
Merge pull request #6 from zhangxuan97/branch-A-CodeQuality

coding quality updates

zhangxuan97 committed:
More SLAP, removed unused methods

zhangxuan97 committed:
Add C-archive

zhangxuan97 committed:
Fixed incorrect naming in C-archive

zhangxuan97 committed:
small changes

zhangxuan97 committed:
Improved helper class

zhangxuan97 committed:
Add CSS styling and fixed issue where height of dialog box is limited.

zhangxuan97 committed:
File path specification hard coding is completely removed.
User should no longer have any issues in loading and writing to task files, an empty one would be created if it does not exist, along with the respective directory.

zhangxuan97 committed:
Add a representative screenshot

zhangxuan97 committed:
Update README

zhangxuan97 committed:
Fix archive out of bounds error

zhangxuan97 committed:
Fix filepath issue

zhangxuan97 committed:
Cross Platform

zhangxuan97 committed:
Update readme: Fix link to release

zhangxuan97 committed:
Change Duke font

zhangxuan97 committed:
Update README: Add section on where external text files are saved

zhangxuan97 committed:
Set theme jekyll-theme-cayman

zhangxuan97 committed:
Fix error where Helvetica Neue font causes cutoff in dialog box.

zhangxuan97 committed:
Fix checkstyle

zhangxuan97 committed:
Fix checkstyle

zhangxuan97 committed:
Update readme
